### PR TITLE
Add validate-graphql-request policy support

### PR DESCRIPTION
**Pull request by tomkerkhove**
## Summary

Adds full support for the \alidate-graphql-request\ policy to the Policy Toolkit.

## Changes

- **Config**: \ValidateGraphqlRequestConfig\, \AuthorizeConfig\, \AuthorizeRuleConfig\ records in \src/Authoring/Configs/\
- **Compiler**: \ValidateGraphqlRequestCompiler\ handles nested \<authorize>\ with \<rule>\ child elements
- **Decompiler**: \ValidateGraphqlRequestDecompiler\ round-trips XML back to C# config
- **IInboundContext**: Added \ValidateGraphqlRequest(ValidateGraphqlRequestConfig config)\ method
- **Tests**: 5 DataRow test cases covering all attributes, authorize rules, and edge cases

## Policy Reference

[validate-graphql-request](https://learn.microsoft.com/en-us/azure/api-management/validate-graphql-request-policy)

## Supported Attributes

| Attribute | Config Property | Type |
|-----------|----------------|------|
| \rror-variable-name\ | \ErrorVariableName\ | \string?\ |
| \max-depth\ | \MaxDepth\ | \int?\ |
| \max-size\ | \MaxSize\ | \long?\ |
| \max-total-depth\ | \MaxTotalDepth\ | \int?\ |
| \max-complexity\ | \MaxComplexity\ | \int?\ |
| \uthorize > rule\ | \Authorize.Rules[]\ | \AuthorizeRuleConfig[]\ |

Co-authored-by: Copilot <223556219+Copilot@users.noreply.github.com>

### CODE REVIEW

File: src/Authoring/IInboundContext.cs [hunk 1]
```diff
@@ -630,6 +630,15 @@ public interface IInboundContext : IHaveExpressionContext
...
     void ValidateContent(ValidateContentConfig config);
 
+    /// <summary>
+    /// Validates a GraphQL request against configured rules.<br/>
+    /// Compiled to <a href="https://learn.microsoft.com/en-us/azure/api-management/validate-graphql-request-policy">validate-graphql-request</a> policy.
+    /// </summary>
+    /// <param name="config">
+    /// Configuration specifying the validation parameters, including depth limits, size limits, complexity, and authorization rules.
+    /// </param>
+    void ValidateGraphqlRequest(ValidateGraphqlRequestConfig config);
```
Comment (Mielek) [inline src/Authoring/IInboundContext.cs:640]: should we no add this as well to the policy fragment context?

File: src/Compiling/Compiling.csproj [hunk 2]
```diff
@@ -1,49 +1,49 @@
...
-    <ItemGroup>
-        <ProjectReference Include="..\Core\Core.csproj"/>
-    </ItemGroup>
-
-    <ItemGroup>
-        <None Include="..\..\docs\images\pkgicon.png" Pack="true" PackagePath="\"/>
-        <None Include="README.md" Pack="true" PackagePath="\"/>
-    </ItemGroup>
-</Project>
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (Mielek) [inline src/Compiling/Compiling.csproj:1]: revert changes to this file please

File: src/Testing/Testing.csproj [hunk 3]
```diff
@@ -1,50 +1,50 @@
...
-
-    <ItemGroup>
-        <ProjectReference Include="..\Authoring\Authoring.csproj" />
-    </ItemGroup>
-
-    <ItemGroup>
-        <None Include="..\..\docs\images\pkgicon.png" Pack="true" PackagePath="\"/>
-        <None Include="README.md" Pack="true" PackagePath="\"/>
-    </ItemGroup>
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (Mielek) [inline src/Testing/Testing.csproj:1]: revert changes to this file please

File: test/Test.Core/Compiling/ValidateGraphqlRequestTests.cs [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+                    <authorize>
+                        <rule path="/Query/users" action="allow" />
+                    </authorize>
+                </validate-graphql-request>
+            </inbound>
+        </policies>
+        """,
+        DisplayName = "Should compile validate-graphql-request policy with single attribute and authorize"
+    )]
+    public void ShouldCompileValidateGraphqlRequestPolicy(string code, string expectedXml)
```
Comment (Mielek) [inline test/Test.Core/Compiling/ValidateGraphqlRequestTests.cs:338]: Can we add tests which check expressions are inline in the fields which support them?

**Review by Mielek — CHANGES_REQUESTED** [review]
